### PR TITLE
Add linters and fix errors

### PR DESCRIPTION
**Pull request by mlguerrero12**

### CODE REVIEW

File: bond/bond_test.go [hunk 1]
```diff
@@ -315,7 +314,7 @@ var _ = Describe("tuning plugin", func() {
 			By("validating the macs are not duplicated")
 			slave1, err = netlink.LinkByName(Slave1)
 			Expect(err).NotTo(HaveOccurred())
-			slave2, err = netlink.LinkByName(Slave2)
+			slave2, _ = netlink.LinkByName(Slave2)
```
Comment (Eoghan1232) [inline bond/bond_test.go:317]: Should we be doing this instead? why ignore the error?
```suggestion
			slave2, err= netlink.LinkByName(Slave2)
			Expect(err).NotTo(HaveOccurred())
```

**Review by Eoghan1232 — APPROVED** [review]

**Comment by Eoghan1232** [comment]
This PR contains the changes from the other 2 PRs #48 #49

File: bond/bond_test.go (hunk 1 above)
Comment (mlguerrero12) [reply bond/bond_test.go:317]: yes, it's better to check the error indeed. Thank you.

**Comment by mlguerrero12** [comment]
Yes, it was rebased on top of #48 and #49. The idea is to merge those first.

**Comment by Eoghan1232** [comment]
sure @mlguerrero12 , I am happy with #48 and #49.
I'll go ahead with the merge of those.

**Comment by Eoghan1232** [comment]
I suspect this will have merge conflicts too after #49 is resolved and merged.

**Comment by mlguerrero12** [comment]
resolved, thanks.

**Pull request merged**